src/foam/u2/view/RichTextArea.js

```diff
@@ -0,0 +1,329 @@
+      min-width: 50px;
+      min-height: 50px;
+      box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.2);
+      z-index: 1;
+    }
+
+    /* Links inside the dropdown */
+    .dropdown-content a {
+      color: black;
+      padding: 12px 16px;
+      text-decoration: none;
+      display: block;
+    }
+
+    /* Change color of dropdown links on hover */
+    .dropdown-content a:hover {background-color: #ddd}    
+  `,
+
+  properties: [
+    {
+      name: 'document'
+    },
+    {
+      class: 'Boolean',
+      name: 'onKey',
+      attribute: true
+    },
+    {
+      type: 'String',
+      name: 'height',
+      value: '400'
+    },
+    {
+      type: 'String',
+      name: 'width',
+      value: '100%'
+    },
+    {
+      name: 'richActions',
+    },
+    {
+      name: 'fontSizeMenu',
+      postSet: function (_, n) {
+        var self = this;
+        n.onload.sub(function () {
+          n.document = self.frame.el().contentDocument;
+          n.window = self.frame.el().contentWindow;
+        });
+      }
+    },
+    {
+      name: 'justifyMenu',
+      postSet: function (_, n) {
+        var self = this;
+        n.onload.sub(function () {
+          n.document = self.frame.el().contentDocument;
+          n.window = self.frame.el().contentWindow;
+        });
+      }
+    },
+    {
+      name: 'fontFaceMenu',
+      postSet: function (_, n) {
+        var self = this;
+        n.onload.sub(function () {
+          n.document = self.frame.el().contentDocument;
+          n.window = self.frame.el().contentWindow;
+        });
+      }
+    },
+    {
+      name: 'otherActionsMenu',
+      postSet: function (_, n) {
+        var self = this;
+        n.onload.sub(function () {
+          n.document = self.frame.el().contentDocument;
+          n.window = self.frame.el().contentWindow;
+        });
+      }
+
+    },
+    {
+      name: 'frame',
+      postSet: function (_, n) {
+        var self = this;
+        n.onload.sub(function () {
+          self.document = n.el().contentDocument;
+
+          n.el().contentDocument.body.style.whiteSpace = 'pre-wrap';
+          n.el().contentDocument.head.insertAdjacentHTML(
+            'afterbegin',
+            '<style>blockquote{border-left-color:#fff;border-left-style:none;padding-left:1ex;}</style>');
+          n.el().contentDocument.body.style.overflow = 'auto';
+          n.el().contentDocument.body.style.margin = '5px';
+          n.el().contentDocument.body.contentEditable = true;
+          n.el().contentDocument.body.insertAdjacentHTML('beforeend', self.data);
+          n.el().contentDocument.body.addEventListener('input', function () {
+            self.data = n.el().contentDocument.body.innerHTML;
+          }, false);
+
+        });
+      }
+    },
+  ],
+
+  methods: [
+    function initE() {
+      this.SUPER();
+      this
+        .start('div').addClass('foam-flow-Document')
+          .start('iframe', null, this.frame$).addClass('iframeContainer')
```

Comment:
I'm not sure we need to do this in an iframe anymore. Foam1 did it this way because content editable support in a div was scarce but now it's pretty widespread so I think we can simplify our lives a little by making this just a content editable div.

Reply:
Would remove this class, it is not used again. Moved it to RichTextEditor

Reply:
Using div in RichTextEditor. Just ignore RichTextArea.js. Would remove it.